Firestore/core/src/firebase/firestore/model/snapshot_version.cc

```diff
@@ -29,6 +31,17 @@ const SnapshotVersion& SnapshotVersion::None() {
   return kNone;
 }
 
+int64_t SnapshotVersion::ToMicroseconds() const {
+  namespace chr = std::chrono;
+
+  auto second_part =
+      chr::duration_cast<chr::microseconds>(chr::seconds(timestamp_.seconds()));
+  auto nanosecond_part = chr::duration_cast<chr::microseconds>(
+      chr::nanoseconds(timestamp_.nanoseconds()));
+  auto result = second_part + nanosecond_part;
```

Comment:
I don't care strongly, but I like the JS one-liner that I could write with my eyes closed. :-)

```
return this.timestamp.seconds * 1e6 + this.timestamp.nanoseconds / 1000;
```

But starting at this method made me question why we're using micros anyway...  I don't think the backend guarantees that read times will never use nano-precision, so this is theoretically a lossy conversion.  That's fine for your delta computation, but it seems slightly strange / dangerous to expose a lossy conversion on SnapshotVersion.

Can you ditch this and just use snapshotVersion.timestamp().seconds() in your delta computation?

Reply:
Out-of-band comment: is there a reason not to use `Timestamp::ToTimePoint` (it's unavailable if STLPort is used, but `<chrono>` would be unavailable as well)?

```cpp
auto microseconds = chr::duration_cast<chr::microseconds>(timestamp_.ToTimePoint().time_since_epoch());
return static_cast<int64_t>(microseconds.count());
```

Reply:
That comment is why I tagged you as a reviewer :-).

Reply:
I did this because the typescript client already had it and the other clients didn't. I think the typescript client has it for generating the spec test json though.

Given the interval is minutes, just using seconds makes sense to me. Removed.